events/api.py

```diff
@@ -3271,9 +3274,8 @@ def _filter_event_queryset(queryset, params, srs=None):  # noqa: C901
         )
 
     if end:
-        dt = utils.parse_time(end, is_start=False)[0]
-        if not dt.tzinfo:
-            dt = dt.astimezone(pytz.timezone("UTC"))
+        # Inconsistent behaviour, see test_inconsistent_tz_default
+        dt = utils.parse_end_time(end, default_tz=pytz.timezone(settings.TIME_ZONE))[0]
```

Comment:
@tuomas777 any thoughts on this? TL;DR create and list event use different tz for naive date(times). List localizes to settings.TIME_ZONE and create sets them to UTC.